src/core/shape/2d_primitives.js

```diff
@@ -374,6 +374,14 @@ p5.prototype.point = function() {
 p5.prototype.quad = function() {
   p5._validateParameters('quad', arguments);
 
+  // if 3D and we weren't passed 12 args, assume Z is 0
+  if (this._renderer.isP3D && arguments.length !== 12) {
+    Array.prototype.splice.call(arguments, 2, 0, 0);
+    Array.prototype.splice.call(arguments, 5, 0, 0);
+    Array.prototype.splice.call(arguments, 8, 0, 0);
+    Array.prototype.push.call(arguments, 0);
+  }
```

Comment:
is this the best way to alter `arguments`?

Reply:
Hi @unjust This is a good start! The only issue is that Array.prototype.slice creates a new array object every time it is called ([the removed elements are returned or if nothing is removed an empty array is returned](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)). I think that you can safely do something such as:
```
this._renderer.quad.call(
  this._renderer,
  arguments[0], arguments[1], 0, 
  arguments[2], arguments[3], 0, 
  arguments[4], arguments[5], 0, 
  arguments[6], arguments[7], 0
)
```
If I am not mistaken you don't need to modify the arguments array. Also, this maybe makes it easier for someone else to quickly understand what it is that you are doing.

Reply:
thanks ok! not sure why I didn't think of that.

Reply:
ok I was altering them because this line in turn uses `arguments`

``` if (this._renderer._doStroke || this._renderer._doFill) {
    this._renderer.quad.apply(this._renderer, arguments);
  }```

I'm not too sure what that line does, do I need to think about the call that's happening there?

Reply:
I understand the confusion. `call` and `apply` are similar. [Here is a short article that explains the difference.](http://www.syntaxsuccess.com/answer-question/556364e45bd0419c3c4cecea/difference-between-call-and-apply). In my opinion, it is easier to read if you use `call` instead of modifying the existing array and using `apply`. Then you can check if `_renderer.isP3D` is true, and if it is then either use this line above `quad.apply(this._renderer, arguments` or the line I suggested depending on whether the `arguments.length` is 12 or 8 respectively. Does that make sense?

Reply:
That makes sense. 
Honestly I overlooked your change from `apply` -> `call` the first time - so I get what you are saying now. Let me know if this is better.

also is there any other way communicate on the p5 project or just through @ mentions?